src/components/Modal/SettingsCacheClearModal/SettingsCacheClearModal.vue

```diff
@@ -0,0 +1,213 @@
+const getStorageUsage = async () => {
+  if (!isStorageManagerSupported) return null
+
+  return navigator.storage.estimate()
+}
+
+const confirmClear = () => window.confirm('本当に削除しますか？')
+
+/* CacheStorageのnameはsw.jsを参照 */
+const clearCacheStorage = (cacheName: string) => window.caches.delete(cacheName)
+</script>
+
+<script lang="ts" setup>
+import { computed } from 'vue'
+import ModalFrame from '../Common/ModalFrame.vue'
+import FormButton from '/@/components/UI/FormButton.vue'
+import { useModalStore } from '/@/store/ui/modal'
+import FormCheckbox from '/@/components/UI/FormCheckbox.vue'
+
+const { addSuccessToast } = useToastStore()
+const showToast = (extraMessage?: string) => {
+  addSuccessToast(
+    `削除に成功しました${extraMessage ? `: ${extraMessage}` : ''}`
+  )
+}
+
+const cacheData = ref<StorageEstimate | null>(null)
+const setCacheData = async () => {
+  cacheData.value = await getStorageUsage()
+}
+onMounted(setCacheData)
+
+type CacheName = 'caches' | 'indexedDB' | 'serviceWorkerRegistrations'
+const caches = 'caches'
+const indexedDB = 'indexedDB'
+const serviceWorkerRegistrations = 'serviceWorkerRegistrations'
+
+const selectedCaches = ref<Record<string, boolean>>({
+  caches: false,
+  indexedDB: false,
+  serviceWorkerRegistrations: false
+})
+const allCachesSelected = ref<boolean>(false)
+const anyCacheSelected = computed(() => {
+  return allCachesSelected.value || Object.values(selectedCaches).find(v => v)
+})
+
+const cacheLabel = (cacheName: string) => {
+  switch (cacheName) {
+    case caches:
+      return 'traQ本体'
+    case indexedDB:
+      return 'ファイルの本体一覧'
+    case serviceWorkerRegistrations:
+      return 'ファイルのサムネイル一覧'
+    default:
+      throw new Error(`Unknown cache name: ${cacheName}`)
+  }
+}
+
+const { clearModal } = useModalStore()
+
+const clearMainCache = async () => {
+  const names = await window.caches.keys()
+  await Promise.all(
+    names
+      .filter(name => name.startsWith('traQ_S-precache'))
+      .map(name => clearCacheStorage(name))
+  )
+}
+
+const clearCache = async () => {
+  if (!confirmClear()) return
+  clearModal()
+  const promises = []
+  if (allCachesSelected.value || selectedCaches.value[caches]) {
+    promises.push(clearMainCache())
+  }
+  if (allCachesSelected.value || selectedCaches.value[indexedDB]) {
+    promises.push(clearCacheStorage('files-cache'))
+  }
+  if (
+    allCachesSelected.value ||
+    selectedCaches.value[serviceWorkerRegistrations]
+  ) {
+    promises.push(clearCacheStorage('thumbnail-cache'))
+  }
+  await Promise.all(promises)
+  if (allCachesSelected.value || selectedCaches.value[caches]) {
+    const registration = await navigator.serviceWorker?.getRegistration()
+    if (!registration) {
+      showToast()
+      return
+    }
+    registration.unregister()
+    showToast('1秒後にリロードします')
+    setCacheData()
+    await wait(1000)
+    window.location.reload()
+    return
+  }
+  setCacheData()
+  showToast()
```

Comment:
早期returnはこういうのを意図していました
```suggestion
if (!(allCachesSelected.value || selectedCaches.value['caches'])) {
    setCacheData()
    showToast()
    return
  }
  const registration = await navigator.serviceWorker?.getRegistration()
  if (registration) {
    registration.unregister()
    showToast('1秒後にリロードします')
    setCacheData()
    await wait(1000)
    window.location.reload()
  } else {
    showToast()
  }
```

Reply:
変えてみました、どうでしょうか